### PR TITLE
[CP-1124] I can delete a single thread from the thread dropdown

### PR DESCRIPTION
**Pull request by wercix**
Jira: [CP-1124]

**Description**
Deleting one thread with deleting, success and fail state
<details>
<summary><b>Screenshots</b></summary>

Screen Shot 2022-05-13 at 02 04 17

Screen Shot 2022-05-12 at 21 44 53

Screen Shot 2022-05-12 at 18 23 53

</details>

[CP-1124]: https://appnroll.atlassian.net/browse/CP-1124?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/messages/reducers/messages.interface.ts [hunk 1]
```diff
@@ -66,6 +66,7 @@ export type MessagesState = Readonly<{
   threadsState: ResultState
   messagesStateMap: { [id: string]: ResultState }
   error: Error | string | null
+  deletingState: ThreadDeletingState | undefined
```
Comment (dkarski) [inline packages/app/src/messages/reducers/messages.interface.ts:69]: To be coherent with the whole project.
```suggestion
  deletingState: ThreadDeletingState | null
```

`undefined` sounds more like an optional variable, `null` meaning that the property was forced as `false value`.

File: packages/app/src/messages/components/messages/messages.component.tsx [hunk 2]
```diff
@@ -144,6 +157,19 @@ const Messages: FunctionComponent<Props> = ({
   const { selectedRows, allRowsSelected, toggleAll, resetRows, ...rest } =
     useTableSelect<Thread>(threads)
 
+  const [deletedThreads, setDeletedThreads] = useState<string[]>([])
+
+  useEffect(() => {
+    if (deletingState === ThreadDeletingState.Success) {
+      const timeout = setTimeout(() => {
+        closeModal()
```
Comment (dkarski) [inline packages/app/src/messages/components/messages/messages.component.tsx:165]: Maybe clear will sounds more generic from the view perspective?
```suggestion
        clearThreadDeletingState()
```

**Comment by dkarski** [comment]
please send files with dedicate OS version for manual tests 🙏 😉

File: packages/app/src/contacts/components/import-contacts-flow/import-contacts-flow.component.tsx [hunk 3]
```diff
@@ -15,7 +15,7 @@ import { defineMessages } from "react-intl"
 import ContactImportModal, {
   ModalType,
 } from "App/contacts/components/contact-import/contact-import-modal.component"
-import ErrorModal from "App/contacts/components/error-modal/error-modal.component"
+import ErrorModal from "App/renderer/components/rest/error-modal/error-modal.component"
```
Comment (igorbezsmertnyi) [inline packages/app/src/contacts/components/import-contacts-flow/import-contacts-flow.component.tsx:18]: We won't to use the `renderer` directory anymore

Comment (dkarski) [reply packages/app/src/contacts/components/import-contacts-flow/import-contacts-flow.component.tsx:18]: `App/ui` is a right place 👍

**Comment by wercix** [comment]
> please send files with dedicate OS version for manual tests 🙏 😉

It can be today's daily ;)

File: packages/app/src/contacts/components/import-contacts-flow/import-contacts-flow.component.tsx (hunk 3 above)
Comment (wercix) [reply packages/app/src/contacts/components/import-contacts-flow/import-contacts-flow.component.tsx]: Fixed in [c415197](https://github.com/mudita/mudita-center/pull/715/commits/c415197ca7813bd72df51545bd3ab908a237534d)

File: packages/app/src/messages/reducers/messages.interface.ts (hunk 1 above)
Comment (wercix) [reply packages/app/src/messages/reducers/messages.interface.ts]: Fixed in [c415197](https://github.com/mudita/mudita-center/pull/715/commits/c415197ca7813bd72df51545bd3ab908a237534d)

File: packages/app/src/messages/messages.container.tsx [hunk 4]
```diff
@@ -65,14 +67,16 @@ const mapDispatchToProps = (dispatch: TmpDispatch) => ({
...
+  deleteThreads: async (threadIds: string[]): Promise<string[] | undefined> =>
+    dispatch(deleteThreads(threadIds)),
   markAsRead: (threadIds: string[]) => dispatch(markThreadsAsRead(threadIds)),
   toggleReadStatus: (threadIds: string[]) =>
     dispatch(toggleThreadsReadStatus(threadIds)),
   addNewMessage: async (newMessage: NewMessage): Promise<Message | undefined> =>
     dispatch(addNewMessage(newMessage)),
   removeLayoutNotification: (notificationId: string) =>
     dispatch(removeNotification(notificationId)),
+  closeModal: () => dispatch(hideDeleteModal()),
```
[the remark this answers is not in the document]
Comment (wercix) [reply packages/app/src/messages/messages.container.tsx]: Fixed in [c415197](https://github.com/mudita/mudita-center/pull/715/commits/c415197ca7813bd72df51545bd3ab908a237534d)

File: packages/app/src/messages/actions/delete-threads.action.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    const { error } = await deleteThreadsRequest(ids)
+    if (error && error.data === undefined) {
+      return rejectWithValue(
+        new DeleteThreadError("Delete Thread request failed")
+      )
+    }
+
+    if (error && error.data !== undefined) {
+      const successIds = ids.filter((id) => error.data.includes(id))
+      return successIds
```
[the remark this answers is not in the document]
Comment (wercix) [reply packages/app/src/messages/actions/delete-threads.action.ts:23]: Tested in [c415197](https://github.com/mudita/mudita-center/pull/715/commits/c415197ca7813bd72df51545bd3ab908a237534d)

File: packages/app/src/messages/components/messages/messages.component.tsx (hunk 2 above)
Comment (wercix) [reply packages/app/src/messages/components/messages/messages.component.tsx]: Here `closeModal` changed to `hideDeleteModal`

**Review by dkarski — APPROVED** [review]

**Review by igorbezsmertnyi — APPROVED** [review]

**Pull request merged**